src/Conjure/UI/Model.hs

```diff
@@ -1014,6 +1014,27 @@ removeExtraSlices model = do
     statements <- descendBiM onExpr (mStatements model)
     return model { mStatements = statements }
 
+lexSingletons :: Monad m => Model -> m Model
+lexSingletons model = do
+  let onExpr :: Monad m => Expression -> m Expression
+      onExpr [essence| &l <lex &r |] =
+        case (matchSingleton l, matchSingleton r) of
+          (Nothing, Nothing) -> return [essence| &l <lex &r |]
+          (Just ls, Just rs) -> return [essence| &ls < &rs |]
+          _ -> bug $ "lexSingleton: match inconsistent" 
+      onExpr [essence| &l <=lex &r |] =
+        case (matchSingleton l, matchSingleton r) of
+          (Nothing, Nothing) -> return [essence| &l <=lex &r |]
+          (Just ls, Just rs) -> return [essence| &ls <= &rs |]
+          _ -> bug $ "lexSingleton: match inconsistent" 
+      onExpr x = return x
+      matchSingleton :: Expression -> Maybe Expression
+      matchSingleton (AbstractLiteral (AbsLitMatrix _ [s])) = Just s
+      matchSingleton (Constant (ConstantAbstract (AbsLitMatrix _ [s]))) =
+        Just (Constant s)
+      matchSingleton _ = Nothing
+  statements <- descendBiM onExpr (mStatements model)
```

Comment:
This has been changed to transformBiM which is what we want. matchSingletons is tidied up with pattern guards. Running the full tests now

Reply:
Looks like I'm going to be able to revert a lot of the expected outputs back to what they are on master after this! Hopefully we can get that 1,471 files changed number down a little :)

Reply:
Sounds good :)

The Azure thing is running all of the tests by the way. I keep failing due to a "lost connection" but for MacOS but if you check the status of the Linux build that should be reliable.

Reply:
Cool. I'll have a look. Down to 130 files changed!